### PR TITLE
Enabling the Tioga adaptive hole map option for the hybrid solver

### PR DESCRIPTION
**Pull request by itopcuoglu**

### CODE REVIEW

File: app/exawind/exawind.cpp [hunk 1]
```diff
@@ -204,6 +204,10 @@ int main(int argc, char** argv)
     const int nonlinear_its = node["nonlinear_iterations"]
                                   ? node["nonlinear_iterations"].as<int>()
                                   : 1;
+    const bool holemap_alg = node["use_adaptive_holemap"]
+                                 ? node["use_adaptive_holemap"].as<bool>()
+                                 : 0;
```
Comment (marchdf) [inline app/exawind/exawind.cpp:209]: change to `: false;` ?

File: src/OversetSimulation.h [hunk 2]
```diff
@@ -99,6 +103,12 @@ class OversetSimulation
         m_nw_start_rank = start_ranks;
         m_num_nw_solvers = m_nw_start_rank.size();
     }
+
+    void set_holemap_alg(int alg)
```
Comment (marchdf) [inline src/OversetSimulation.h:107]: here you are taking an int, but it comes in as a bool, but you use it as an int.

File: src/OversetSimulation.h [hunk 3]
```diff
@@ -32,6 +32,10 @@ class OversetSimulation
     int m_last_timestep{0};
     //! Flag indicating whether initialization tasks have been performed
     bool m_initialized{false};
+    //! Flag indicating if complementary comms have been initialized
+    bool m_comp_comm_initialized{false};
```
Comment (marchdf) [inline src/OversetSimulation.h:36]: can you call this `m_complementary_comm_etc` ? just so the name is a bit more explicit

File: app/exawind/exawind.cpp (hunk 1 above)
Comment (itopcuoglu) [reply app/exawind/exawind.cpp:209]: I will change this accordingly.

File: src/OversetSimulation.h (hunk 2 above)
Comment (itopcuoglu) [reply src/OversetSimulation.h:107]: Alright. I think there is automatic casting, but I should probably just explicitly cast the boolean to an int inside `set_holemap_alg`.

File: src/OversetSimulation.h (hunk 3 above)
Comment (itopcuoglu) [reply src/OversetSimulation.h:36]: Will do.

File: src/OversetSimulation.h (hunk 2 above)
Comment (marchdf) [reply src/OversetSimulation.h:107]: Or just keep it as an int from the start

File: app/exawind/exawind.cpp (hunk 1 above)
Comment (itopcuoglu) [reply app/exawind/exawind.cpp]: I made this change in the update.

File: src/OversetSimulation.h (hunk 2 above)
Comment (itopcuoglu) [reply src/OversetSimulation.h]: With the latest changes, the yaml file reads in a boolean, and an integer of 1 is passed to `set_holemap_alg` if the boolean is true.

File: src/OversetSimulation.h (hunk 3 above)
Comment (itopcuoglu) [reply src/OversetSimulation.h]: This has been changed to `m_complementary_comm_initialized`.

File: src/OversetSimulation.cpp [hunk 4]
```diff
@@ -93,6 +93,10 @@ void OversetSimulation::perform_overset_connectivity()
     m_timers_tg.tick("Connectivity");
     if (m_has_amr) m_tg.preprocess_amr_data();
     m_tg.profile();
+    if (m_holemap_alg == 1 && m_complementary_comm_initialized == false) {
```
Comment (marchdf) [inline src/OversetSimulation.cpp:96]: would you mind doing `((m_holemap_alg == 1) && (m_complementary_comm_initialized == false))` for clarity?

Comment (itopcuoglu) [reply src/OversetSimulation.cpp]: I just pushed this change.

File: app/exawind/exawind.cpp [hunk 5]
```diff
@@ -204,6 +204,10 @@ int main(int argc, char** argv)
     const int nonlinear_its = node["nonlinear_iterations"]
                                   ? node["nonlinear_iterations"].as<int>()
                                   : 1;
+    const bool holemap_alg = node["use_adaptive_holemap"]
```
Comment (marchdf) [inline app/exawind/exawind.cpp:207]: `holemap_alg` can we rename to `adaptive_holemap_alg`

File: src/OversetSimulation.h [hunk 6]
```diff
@@ -99,6 +103,12 @@ class OversetSimulation
         m_nw_start_rank = start_ranks;
         m_num_nw_solvers = m_nw_start_rank.size();
     }
+
+    void set_holemap_alg(bool alg)
+    {
+        m_holemap_alg = alg;
```
Comment (marchdf) [inline src/OversetSimulation.h:109]: `m_holemap_alg` to `m_is_adaptive_holemap_alg`

**Review by marchdf — APPROVED** [review]

**Pull request merged**